src/bulk/common.ts

```diff
@@ -206,6 +206,21 @@ export class BulkWriteResult {
     this.upsertedCount = this.result.upserted.length ?? 0;
     this.upsertedIds = BulkWriteResult.generateIdMap(this.result.upserted);
     this.insertedIds = BulkWriteResult.generateIdMap(this.result.insertedIds);
+    if (isOrdered && this.result.writeErrors.length !== 0) {
```

Comment:
It seems like this logic might be a bit easier to understand if instead of 1) constructing the id map and then 2) deleting unnecessary values from it, if we just pre-filtered `insertedIds` before constructing the object

Breaking apart the logic makes it really clear what's happening in both cases

- for ordered writes, all insertedIds before the error are valid
- for unordered writes, we filter out all the errors

```typescript
function getInsertedIds(bulkResult: BulkResult, isOrdered: boolean) {
  if (bulkResult.writeErrors.length === 0) return bulkResult.insertedIds;

  if (isOrdered) {
    return bulkResult.insertedIds.slice(0, bulkResult.writeErrors[0].index);
  }

  const writeErrors = bulkResult.writeErrors;

  return bulkResult.insertedIds
	.filter(({ index }) => !writeErrors.includes(writeError => index === writeError.index))
}

class BulkWriteResult {
  constructor(..) {
    ...
    this.insertedIds = BulkWriteResult.generateIdMap(
      getInsertedIds(bulkResult, isOrdered)
    )
  }
}
```